crates/hotshot/src/lib.rs

```diff
@@ -637,6 +643,15 @@ pub struct HotShotInitializer<TYPES: NodeType> {
 
     /// Starting view number that we are confident won't lead to a double vote after restart.
     start_view: TYPES::Time,
+    /// Highest QC that was seen, for genesis it's the genesis QC.  It should be for a view greater
+    /// than `inner`s view number for the non genesis case because we must have seen higher QCs
+    /// to decide on the leaf.
+    high_qc: QuorumCertificate<TYPES>,
+    /// Undecided leafs that were seen, but not yet decided on.  These allow a restarting node
+    /// to vote and propose right away if they didn't miss anything while down.
+    undecided_leafs: Vec<Leaf<TYPES>>,
+    /// Not yet decided state
+    undecided_state: BTreeMap<TYPES::Time, View<TYPES>>,
```

Comment:
Just for my own clarity: The BTree map here is to ensure that the highest view is always at the top?

Reply:
For the most part it just makes the garbage collection easier.  In this case it's the same because it's what hotshot already has available.

We store it in memory as a `BTreeMap` because we only want to keep the undecided views.  When we get decide we can easily remove all views <= the decided view from the map thanks to the ordering.